web/impact/impact/views/algolia_api_key_view.py

```diff
@@ -19,23 +21,38 @@ class AlgoliaApiKeyView(APIView):
     actions = ["GET"]
 
     def get(self, request, format=None):
+        index_prefix = settings.ALGOLIA_INDEX_PREFIX
         client = algoliasearch.Client(
             settings.ALGOLIA_APPLICATION_ID,
             settings.ALGOLIA_API_KEY)
         params = {
             'hitsPerPage': 24,
-            'filters': '',
             'validUntil': int(time.time()) + 3600,
             'userToken': request.user.id
         }
         search_key = settings.ALGOLIA_SEARCH_ONLY_API_KEY
         if request.user.is_staff:
             search_key = settings.ALGOLIA_STAFF_SEARCH_ONLY_API_KEY
-
+        else:
+            active_roles = UserRole.FINALIST_USER_ROLES
+            active_roles.append(UserRole.MENTOR)
+            facet_filters = []
+            for grant in request.user.programrolegrant_set.filter(
+                    program_role__program__program_status__in=CURRENT_STATUSES,
+                    program_role__user_role__name__in=active_roles
+            ):
+                facet_filters.append(
+                    'confirmed_mentor_programs:"{active_program}"'.format(
+                        active_program=grant.program_role.program.name))
+            if len(facet_filters) > 0:
```

Comment:
the next 4 lines seem very weird to me, why would `"is_confirmed_mentor:true"` be referenced as a filter only if the facets are empty?

Reply:
Let me guess - Algolia won't create the facets if the different programs won't be separated by OR?
The current implementation is very confusing.

Reply:
this was done so that for regular non-staff users with an active program will see only relevant confirmed mentors for their program and non-staff users not in an active program see all confirmed mentors for all programs. This behavior was fallout from a discussion in an exec meeting but will defer to @EcoLek on wether or not this is desirable. The alternative would be to show either no results for these users or only mentors from their last active program.